File: Amathus/Amathus.Reader/Common/Feeds/FeedItem.cs
using System;
using Newtonsoft.Json;

namespace Amathus.Reader.Common.Feeds
{
    public class FeedItem
    {
        public string Title { get; set; }

        public DateTime PublishDate { get; set; }

        public string Summary { get; set; }

        public string Detail { get; set; }

        public Uri ImageUrl { get; set; }

        public Uri Url { get; set; }

        [JsonIgnore]
        public int Length
        {
            get
            {
                return GetLength(Title) + GetLength(Summary) + GetLength(Detail);
            }
        }

        private static int GetLength(string key)
        {
            return key?.Length ?? 0;
            //return key == null ? 0 : key.Length;
        }
    }
}

File: AmathusClient/lib/models/feeditem.dart
import 'package:amathus/models/feed.dart';
import 'package:json_annotation/json_annotation.dart';

// Run the following to generate it:
// flutter packages pub run build_runner build
part 'feeditem.g.dart';

@JsonSerializable()
class FeedItem {

  @JsonKey(name: 'Title')
  final String title;

  @JsonKey(name: 'PublishDate')
  final DateTime publishDate;

  @JsonKey(name: 'Summary')
  final String summary;

  @JsonKey(name: 'Detail')
  final String detail;

  @JsonKey(name: 'ImageUrl')
  final String imageUrl;

  @JsonKey(name: 'Url')
  final String url;

  @JsonKey(ignore: true)
  Feed feed;

  FeedItem(this.title, this.publishDate, this.summary, this.detail,
      this.imageUrl, this.url);

  factory FeedItem.fromJson(Map<String, dynamic> json) => _$FeedItemFromJson(json);

  Map<String, dynamic> toJson() => _$FeedItemToJson(this);
}